syntax = "proto3";

package weatherService;

service WeatherService {
    rpc GetCurrentWeather (WeatherRequest) returns (WeatherResponse);
    rpc GetWeatherPrediction (WeatherPredictionRequest) returns (WeatherPredictionResponse);

}

message WeatherRequest {
    string location = 1;
}

message WeatherResponse {
    string location = 1;
    string weather = 2;
    float temperature = 3;
    float wind_speed = 4;
}

message WeatherPredictionRequest {
    string location = 1;
}


message WeatherPredictionResponse {
    string location = 1;
    string forecast_date = 2;
    float predicted_temperature = 3;
    float predicted_wind_speed = 4;
    float predicted_precipitation = 5;
    string prediction_generated_at = 6;
}